### PR TITLE
Feature: provide hostAppInfo to user-agent

### PR DESCRIPTION
**Pull request by minh-rakuten**
# Description
Allow sdk to get info from host app for analytics, app traffic, ads. This append provided host app info to user-agent of webview.
Update the user-agent for Android Demo App to append "MiniApp Demo App" for Google Analytics.

## Links
[MINI-1144](https://jira.rakuten-it.com/jira/browse/MINI-1144)

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt [hunk 1]
```diff
@@ -20,6 +20,11 @@ abstract class MiniAppMessageBridge {
         callback: (isGranted: Boolean) -> Unit
     )
 
+    /** This host app info can apply to analytics, app traffic, ads, etc.
+     *  It should be a meaningful keyword such as host app name to differentiate other host apps.
+     **/
+    abstract fun getHostAppInfo(): String
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt:26]: I'm not sure if `MiniAppMessageBridge` is the right place for this. This class is intended for sending data between the host app and mini app, but the `hostAppInfo` isn't accessible from the Mini App.

I was thinking, maybe this could instead be an optional setting in the manifest meta-data? If the host app doesn't use the setting, then the SDK would automatically retrieve the App name using `ApplicationInfo`. What do you think @minh-rakuten @ts-nitish-a-jain ?

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt [hunk 2]
```diff
@@ -20,6 +20,11 @@ abstract class MiniAppMessageBridge {
         callback: (isGranted: Boolean) -> Unit
     )
 
+    /** This host app info can apply to analytics, app traffic, ads, etc.
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt:23]: I think the documentation needs to mention that this is being appended to the User Agent.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt (hunk 1 above)
Comment (minh-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt:26]: I'm on it.

Comment (ts-nitish-a-jain) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt]: Correct. 
In slack thread also, it was aligned, a few days back I think, that this needs to be a build time setting.

When not provided, this should be default value and no changes.
When provided, appended to the userAgentString.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniappSdkInitializer.kt [hunk 3]
```diff
@@ -58,16 +65,28 @@ class MiniappSdkInitializer : ContentProvider() {
...
+                isTestMode = manifestConfig.isTestMode()
+            )
         )
 
         return true
     }
 
+    @VisibleForTesting
+    internal fun provideHostAppInfo(context: Context, manifestConfig: AppManifestConfig): String =
+        if (manifestConfig.hostAppInfo().isEmpty()) {
```
Comment (ts-nitish-a-jain) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniappSdkInitializer.kt:83]: Please check my other comment. 
It needs to be explicitly set by consumer app as it is an entity for the host app's identification (and can involve more params like version etc.)

Comment (minh-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniappSdkInitializer.kt:83]: Okay. I will not add the default string here.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniappSdkInitializer.kt [hunk 4]
```diff
@@ -58,16 +65,18 @@ class MiniappSdkInitializer : ContentProvider() {
...
+                subscriptionKey = manifestConfig.subscriptionKey(),
+                hostAppVersionId = manifestConfig.hostAppVersion(),
+                hostAppInfo = manifestConfig.hostAppInfo(),
+                isTestMode = manifestConfig.isTestMode()
+            )
         )
 
         return true
     }
-
```
Comment (khairul-alam-licon) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniappSdkInitializer.kt:70]: Unnecessary change?

File: miniapp/USERGUIDE.md [hunk 5]
```diff
@@ -27,9 +27,23 @@ dependency {
...
 
-### #2 Set your App Id, Subscription Key, & Base URL
+### #2 Configure SDK settings in AndroidManifest.xml
 
-We don't currently host a public API, so you will need to provide your own Base URL for API requests.
+The SDK is configured via manifest meta-data, the configurable values are:
+
+| Field                        | Datatype| Manifest Key                                         | Optional   | Default   |
+|------------------------------|---------|------------------------------------------------------|------------|---------- |
+| Base URL                     | String  | `com.rakuten.tech.mobile.miniapp.BaseUrl`            | ❌         | 🚫        |
```
Comment (khairul-alam-licon) [inline miniapp/USERGUIDE.md:36]: What is the difference between ❌ and 🚫 ?

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebView.kt [hunk 6]
```diff
@@ -38,6 +39,11 @@ internal class MiniAppWebView(
         settings.allowUniversalAccessFromFileURLs = true
         settings.domStorageEnabled = true
         settings.databaseEnabled = true
+
+        if (hostAppInfo.isNotEmpty())
+            settings.userAgentString =
+                String.format("%s %s", settings.userAgentString, hostAppInfo)
```
Comment (khairul-alam-licon) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebView.kt:45]: Does this UA string should be unique?

Comment (minh-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebView.kt:45]: No, it does not need to be unique. The specs only require the `hostAppInfo` would be appended.

File: miniapp/USERGUIDE.md [hunk 7]
```diff
@@ -27,9 +27,23 @@ dependency {
...
+| Testbox Config               | boolean | `com.rakuten.tech.mobile.miniapp.IsTestMode`         | ✅         | `false`   |
+| Host App Version             | String  | `com.rakuten.tech.mobile.miniapp.HostAppVersion`     | ❌         | 🚫        |
+| Host App Info                | String  | `com.rakuten.tech.mobile.miniapp.HostAppInfo`        | ✅         | 🚫        |
```
Comment (ts-nitish-a-jain) [inline miniapp/USERGUIDE.md:39]: This needs to be updated to be reflective of what this information is specifically being used for. Hence,  
`Host App Info` ->  `Host App User Agent Info`

File: testapp/build.gradle [hunk 8]
```diff
@@ -36,13 +36,15 @@ android {
     def stagingManifestPlaceholders = [
             baseUrl        : property("MINIAPP_SERVER_BASE_URL") ?: "https://www.example.com/",
             isTestMode     : property("IS_TEST_MODE") ?: false,
+            hostAppInfo    : property("HOST_APP_INFO") ?: "MiniApp Demo App",
```
Comment (ts-nitish-a-jain) [inline testapp/build.gradle:39]: A more analogous user agent info could be inferred from the existing user agent information for our sample app. What do you think?
